### ISSUE TITLE
Activate objed with object at point when the point is at boundaries

### ISSUE THREAD

**Issue by Wetlize** [clemera/objed#68]
I've never found intuitive moving to the next/previous object after activating objed with `objed-end-of-object-at-point`/`objed-beg-of-object-at-point` when the point is located at the object's boundary. It is quite opposite, most of the time I am ending up annoyed when this happens.

What I would like it to do instead, is always select object at point (like it does right now if the point is not at the boundaries) and only after repeat, it would move to the next/previous object. Additionally, the user could pass `universal-argument`/`digit-argument` to explicitly require movement to the next/previous object ahead of time, when this is really desirable.

The current behaviour of these commands can also be preserved (as right now) when objed is already active. Though I don't have a good opinion about this since I use these commands very rarely when objed is already active.

**Comment by clemera** [comment]
Thanks, I think staying at the object at point when at the boundaries is a good idea! Are there any downsides to stay at the boundaries on repeat? The editing flow would be like: `M-[` or `M-]` to activate the object at point and go to the boundary, then `[` or `]` to navigate to previous/next object.

**Comment by Wetlize** [comment]
> Are there any downsides to stay at the boundaries on repeat (so basically do nothing on repeat)? 
I don't think so, in fact, I've probably never even used `M-[`/`M-]` with already active objed.

> The editing flow would be like: M-[ or M-] to activate the object at point (and go to the boundary if not already there), then [ or ] to navigate to previous/next object.
Yeah, I like this.

**Comment by Wetlize** [comment]
Thanks.

**Comment by Wetlize** [comment]
I feel like the behavior is still not intuitive, consider the next cases (where `|` characters represents point's position):
```emacs-lisp
(defun foo ()
  ...)

(defun bar ()
  ...)  
|
(defun foobar ()
  ...)
```
Hitting `M-[` from here will select `bar` and `M-]` will select `foobar`. Both of these are expected, good. 

```emacs-lisp
(defun foo ()
  ...)

(defun bar ()
  ...)  |

(defun foobar ()
  ...)
```
Here, if you hit `M-[` it will select `foo` for some weird reason, while `M-]` will select `foobar.

```emacs-lisp
(defun foo ()
  ...)

(defun bar ()
  ...)  
|
;; surprise comment

(defun foobar ()
  ...)
```

And here both, `M-[` and `M-]` will select comment.

See the pattern? This is not consistent. Are these considered as bugs? Expected behaviors? Flows in implementations? It is difficult to rely on these commands when they don't behave consistently.

**Comment by Wetlize** [comment]
Quite the same story applies to `objed-until-beg-of-object-at-point` (`M-(`) and `objed-until-beg-of-object-at-point` (`M-)`). But they additionally can put cursor on their own (unique) positions, different from `M-[` and `M-]`.

```emacs-lisp
(defun foo ()
  ...)

(defun bar ()     
  ...)  

;; comment  
|
(d|<efun foobar ()
  ...|>)
```
Here `<|` indicates the position of point for `M-(` and `|>` indicates the position of point for `M-)`. This is the same as the last example, except they put point on different positions.

**Comment by clemera** [comment]
Thanks for the detailed examples, I will use them to improve the behavior. It's mostly because I reused  code that can no be reused for this purpose, I think.
